ftlib/consensus/gossip/impl.py

```diff
@@ -49,7 +49,7 @@ def __init__(
         if known_addr_list is not None and known_addr_list != []:
             joined = self._join(known_addr_list=known_addr_list)
             if not joined:
-                raise RuntimeError("failed to join the group")
+                logging.warning("failed to join the group")
```

Comment:
We should also expose a property like `is_joined` so we know when we need to manually join. Otherwise, `_join()` will need to be called again just to find out whether we have already joined.

Reply:
Right, I’ve exposed the joined status in the new commit.